### PR TITLE
chore(jellyfish-transaction): add static testing and e2e for v2 signing 

### LINKED ISSUE #1366: Transaction V2 format is supported on consensus core but not yet tested
#### What would you like to be added:

This branch contains the basic change to allow Transaction V2 Vout to be signed correctly. https://github.com/DeFiCh/jellyfish/tree/fuxingloh/v2-signing

- [x] Add Static Testing
- [x] Add E2E Testing (submit to testcontainers)

/triage accepted
/area packages

### PR DESCRIPTION
**Pull request by canonbrother**
#### Which issue(s) does this PR fixes?:

Fixes #1366

### CODE REVIEW

**Comment by fuxingloh** [comment]
https://github.com/JellyfishSDK/jellyfish/issues/1366#issuecomment-1141732156

How come?

**Comment by canonbrother** [comment]
> [#1366 (comment)](https://github.com/JellyfishSDK/jellyfish/issues/1366#issuecomment-1141732156)
> 
> How come?

yea i stuck at the v2 vin txid... 
understand the `createrawtx(input, output)` in testcontainer will output an unsigned v4 tx
the way i'm doing is using the bitcoinjs-lib generates a signed v2 tx then just `sendrawtx(signedV2)` (static testing)
but for e2e, i need an actual `vin txid v2` which must be generated from testcontainer... 

hmm... or else that got the other approach?? :thinking:

**Comment by fuxingloh** [comment]
> > [#1366 (comment)](https://github.com/JellyfishSDK/jellyfish/issues/1366#issuecomment-1141732156)
> > How come?
> 
> yea i stuck at the v2 vin txid... understand the `createrawtx(input, output)` in testcontainer will output an unsigned v4 tx the way i'm doing is using the bitcoinjs-lib generates a signed v2 tx then just `sendrawtx(signedV2)` (static testing) but for e2e, i need an actual `vin txid v2` which must be generated from testcontainer...
> 
> hmm... or else that got the other approach?? 🤔

Perhaps we can use testcontainers to use `listunspent` to get `txid, vout` for a particular address. Then manually construct the `V2 Txn` with the given information and sign it with the PrivateKey extracted out from the testcontainers. And resubmit that back into testcontainers?

**Comment by canonbrother** [comment]
> > > [#1366 (comment)](https://github.com/JellyfishSDK/jellyfish/issues/1366#issuecomment-1141732156)
> > > How come?
> > 
> > 
> > yea i stuck at the v2 vin txid... understand the `createrawtx(input, output)` in testcontainer will output an unsigned v4 tx the way i'm doing is using the bitcoinjs-lib generates a signed v2 tx then just `sendrawtx(signedV2)` (static testing) but for e2e, i need an actual `vin txid v2` which must be generated from testcontainer...
> > hmm... or else that got the other approach?? thinking
> 
> Perhaps we can use testcontainers to use `listunspent` to get `txid, vout` for a particular address. Then manually construct the `V2 Txn` with the given information and sign it with the PrivateKey extracted out from the testcontainers. And resubmit that back into testcontainers?

yea.. i was actually going to do that way.. but have mixed feeling to import bitcoinjs-lib and ecpair... 
but now just fix it and push

File: package.json [hunk 1]
```diff
@@ -28,6 +28,8 @@
   },
   "devDependencies": {
     "@types/jest": "^27.4.1",
+    "bitcoinjs-lib": "^6.0.1",
+    "ecpair": "^2.0.1",
```
Comment (fuxingloh) [inline package.json]: should create a new package called `@defichain/bitcoinjs` to separate them

**Comment by fuxingloh** [comment]
Due to the introduction of #1676 where we introduce: "This PR contains no code change on how to use bitcoinjs-lib for DeFiChain without any code changes or forks. The example is taken from https://github.com/fuxingloh/dfi-rawtx-bitcoinjs and to be included into jellyfish repo."

I have updated this PR to follow through with that instead of using the default bitcoinjs client.

Hence, instead of creating `@defichain/bitcoinjs`, I've updated tests in `@defichain/jellyfish-transaction/__tests__` to import the original `bitcoinjs-lib` and test its compatibility instead.

**Comment by fuxingloh** [comment]
> Due to the introduction of #1676 where we introduce: "This PR contains no code change on how to use bitcoinjs-lib for DeFiChain without any code changes or forks. The example is taken from https://github.com/fuxingloh/dfi-rawtx-bitcoinjs and to be included into jellyfish repo."
> 
> I have updated this PR to follow through with that instead of using the default bitcoinjs client.
> 
> Hence, instead of creating `@defichain/bitcoinjs`, I've updated tests in `@defichain/jellyfish-transaction/__tests__` to import the original `bitcoinjs-lib` and test its compatibility instead.

- [x] @canonbrother do take a look and add an "LGTM" if the changes are acceptable. Thanks!

**Comment by canonbrother** [comment]
yea LGTM! :pray:

**Review by fuxingloh — APPROVED** [review]

**Pull request merged**